src/monodroid/jni/monodroid-glue.cc

```diff
@@ -1151,27 +1151,88 @@ MonodroidRuntime::monodroid_dlopen (const char *name, int flags, char **err, [[m
 {
 	unsigned int dl_flags = monodroidRuntime.convert_dl_flags (flags);
 	bool libmonodroid_fallback = false;
-
+	bool name_is_full_path = false;
+	bool name_needs_free = false;
 	/* name is nullptr when we're P/Invoking __Internal, so remap to libxa-internal-api */
-	if (name == nullptr) {
+	if (name == nullptr || strstr (name, "xa-internal-api") != nullptr) {
+#if defined (WINDOWS)
+		char *tmp_name = nullptr;
+
+		auto probe_dll_at = [&](const char *the_path) -> bool {
+			if (the_path == nullptr) {
+				return false;
+			}
+
+			const char *last_sep = strrchr (the_path, MONODROID_PATH_SEPARATOR_CHAR);
+			if (last_sep != nullptr) {
+				char *dir = utils.strdup_new (the_path, last_sep - the_path);
+				tmp_name = utils.string_concat (dir, MONODROID_PATH_SEPARATOR, API_DSO_NAME);
+				delete[] dir;
+				if (!utils.file_exists (tmp_name)) {
+					delete[] tmp_name;
+					tmp_name = nullptr;
+					return false;
+				}
+
+				return true;
+			}
+
+			return false;
+		};
+
+		// First try to see if it exist at the path pointed to by `name`. With p/invokes, currently (Sep 2020), we can't
+		// really trust the path since it consists of *some* directory path + p/invoke library name and it does not
+		// point to the location where the native library is. However, we still need to check the location first, should
+		// it point to the right place in the future.
```

Comment:
TODO (fix after current PR build completes): to link to: https://github.com/mono/mono/issues/20295#issuecomment-679271621